### ISSUE TITLE
Rule Enhancement: `no-extra-bind` arrow function

### ISSUE THREAD

**Issue by ChristianMurphy** [eslint/eslint#3141]
[Arrow functions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions) are permanently Lexically Scoped.
Calling bind on an arrow function has no affect.

Example
``` js
foo = () => {
    console.log( this );
};

// INVALID bind has no effect on `foo`
foo.bind({something: 'else'});
```

Calling bind where it has no effect seems to be a violation of best practice.

ref comments by @IanVS and @ilyavolodin in #3113 
ref [You don't know javascript](https://github.com/getify/You-Dont-Know-JS/blob/master/this%20%26%20object%20prototypes/ch2.md#lexical-this)

**Comment by eslintbot** [comment]
Thanks for the issue! We get a lot of issues, so this message is automatically posted to each one to help you check that you've included all of the information we need to help you.

Reporting a bug? Please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
1. The source code that caused the problem
1. The configuration you're using (for the rule or your entire config file)
1. The actual ESLint output complete with line numbers

Requesting a new rule? Please be sure to include:

1. The use case for the rule - what is it trying to prevent or flag?
1. Whether the rule is trying to prevent an error or is purely stylistic
1. Why you believe this rule is generic enough to be included

Requesting a feature? Please be sure to include:

1. The problem you want to solve (don't mention the solution)
1. Your take on the correct solution to problem

Including this information in your issue helps us to triage it and get you a response as quickly as possible.

Thanks!

**Comment by nzakas** [comment]
@ChristianMurphy what are you asking for? Are you reporting a bug? Requesting a feature? Please see the guidelines above for the information we need.

**Comment by ChristianMurphy** [comment]
Requesting a new feature, in an existing rule.

1. Problem calling bind on an Arrow Function is a violation of best practice.
2. For each arrow function stored to a variable, check if `bind()` is called on the variable.

**Comment by michaelficarra** [comment]
@ChristianMurphy The latter is impossible to do statically. That's entirely a runtime concept.

**Comment by ChristianMurphy** [comment]
@michaelficarra 
It is entirely a runtime concept, because variables can be overwritten at any point in time?
Or is there another reason?
If that is because of overwriting, this could be checked on an arrow function assigned to `const` which guarantees that it cannot be overwritten at run time.
Correct?

**Comment by michaelficarra** [comment]
Also, @nzakas has said previously that he does not want any data flow analysis in rules.

**Comment by ChristianMurphy** [comment]
Without flow analysis, this rule would be kinda useless, closing this issue out.
